rpc/rpcserver.go

```diff
@@ -0,0 +1,170 @@
+package rpc
+
+import (
+	"encoding/hex"
+	"github.com/spacemeshos/post/initialization"
+	"github.com/spacemeshos/post/proving"
+	"github.com/spacemeshos/post/rpc/api"
+	"github.com/spacemeshos/post/shared"
+	"github.com/spacemeshos/post/signal"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var (
+	ErrNotInitialized = status.Error(codes.FailedPrecondition, "not initialized")
+	ErrNoProofExists  = status.Error(codes.FailedPrecondition, "no computed proof exists")
+)
+
+type state struct {
+	id     []byte
+	dir    string
+	proofs map[string]*proving.Proof
+}
+
+// rpcServer is a gRPC, RPC front end to the POST server.
+type rpcServer struct {
+	s       *signal.Signal
+	params  *shared.Params
+	datadir string
+	lograte uint64
+	state   *state
+}
+
+// A compile time check to ensure that rpcServer fully implements PostServer.
+var _ api.PostServer = (*rpcServer)(nil)
+
+// newRPCServer creates and returns a new instance of the rpcServer.
+func NewRPCServer(s *signal.Signal, params *shared.Params, datadir string, lograte uint64) *rpcServer {
+	return &rpcServer{
+		s:       s,
+		params:  params,
+		datadir: datadir,
+		lograte: lograte,
+	}
+}
+
+func (r *rpcServer) Initialize(ctx context.Context, in *api.InitializeRequest) (*api.InitializeResponse, error) {
```

Comment:
Initialize should be non-blocking. It should return once it validated the input.